management/errors.go

```diff
@@ -24,3 +23,14 @@ func IsResourceNotFoundError(err error) bool {
 	azureErr, ok := err.(AzureError)
 	return ok && azureErr.Code == "ResourceNotFound"
 }
+
+// getAzureError converts an error response body into an AzureError instance.
+func getAzureError(responseBody []byte) error {
+	var azErr AzureError
+	err := xml.Unmarshal(responseBody, &azErr)
+	if err != nil {
+		return err
+	}
+
+	return fmt.Errorf("Failed parsing contents to AzureError format: %v", err)
```

Comment:
this will always return err? Comparing with the old code, I think you want to wrap the error on line 32 and return azErr here (currently, azErr is not used at all)...